staging/src/k8s.io/legacy-cloud-providers/aws/aws.go

```diff
@@ -1190,8 +1190,12 @@ func init() {
 			return nil, fmt.Errorf("unable to validate custom endpoint overrides: %v", err)
 		}
 
+		regionName, err := getRegionFromMetadata(cfg)
+		if err != nil {
+			return nil, err
+		}
 		sess, err := session.NewSessionWithOptions(session.Options{
-			Config:            aws.Config{},
+			Config:            *aws.NewConfig().WithRegion(regionName).WithSTSRegionalEndpoint(endpoints.RegionalSTSEndpoint),
```

Comment:
> This is required because, as of SDK v1.42.23, the default is legacy (endpoints.LegacySTSEndpoint).

I can't tell if this is changing behavior from K8s 1.25, or preserving previous behavior.

Reply:
from what i understand it is preserving the previous behavior.

@nckturner @SaranBalaji90 @micahhausler please chime in.

Reply:
if so, it's surprising that the SDK didn't default to preserve previous behavior

Reply:
My understanding is that this overrides the SDK default to use regional endpoints, which is the recommended approach (we made this change in the external cloud provider here: https://github.com/kubernetes/cloud-provider-aws/pull/313).

Reply:
if someone using the in-tree aws provider upgrades from 1.25 to 1.26 that picks up this change, do they need to react at all to this or is it transparent to them? does it matter if they are running in a cluster where the master is on 1.26 with this change and there are 1.25 nodes that don't have this change?

Reply:
Not my area of expertise but apparently its possible to [deactivate regional endpoints], so its possible that sort of setup would be incompatible with the above change (though I haven't tested that). 

[deactivate regional endpoints]: https://docs.aws.amazon.com/IAM/latest/UserGuide/id_credentials_temp_enable-regions.html